core/chains/evm/types/models.go

```diff
@@ -52,8 +55,23 @@ func NewHead(number *big.Int, blockHash common.Hash, parentHash common.Hash, tim
 	}
 }
 
+func (h *Head) BlockNumber() int64 {
```

Comment:
So in summary, this approach extends the evmtypes.Head to implement the txmgrtypes.Head interface

Reply:
Yes